text/0060-refashion-updates-apis.md

```diff
@@ -0,0 +1,429 @@
+- Title: Refashion Updates APIs
+- Start Date: 2021-08-13
+- Specification PR: [#55](https://github.com/meilisearch/specifications/pull/60)
+- Discovery Issue: [#48](https://github.com/meilisearch/product/issues/48)
+
+# Refashion Updates APIs
+
+## 1. Functional Specification
+
+### I. Summary
+
+The term `update` is not the best choice as it can be confused with document updates or `settings`. We have chosen to replace this term with `task`, which is more generic and better reflects the meaning of this API resource.
+
+As an additional change, we have reworked the format of an update to make it more in line with our expectations of an API that is supposed to be easily understandable and developer-oriented.
+
+The changes we will make to the response format of the `task` object lists will make it easier to add more functionality in a future iteration. See the Future Possibilities section for a brief overview.
+
+Two new API endpoints are added. Although quite simple, they allow to consult the list of tasks or a specific task without being forced to know the related index.
+
+#### Summary Key Points
+
+- The `update` resource is renamed `task`. The names of existing API routes are also changed to reflect this change.
+- Tasks are now also accessible as an independent resource of an index. `GET - /tasks`; `GET - /tasks/:taskUid`
+- The task `uid` is not incremented by index anymore. The sequence is generated globally.
```

Comment:
What where the motivation behind this choice? As they make chronology harder to see.

Reply:
Why would the chronology be more difficult to understand in this case? 

The uids are always sorted from the most recent to the oldest. It will always be possible to go to a specific index and have the tasks in a chronological way for a specific index. 

The goal was to provide a global endpoint for teams with several indexes to manage. If we had kept a sequence by index we could have done it with difficulty. 

This should also be a plus when implementing high availability because the sequence of identifiers is globalized in the whole system.

Reply:
> The goal was to provide a global endpoint for teams with several indexes to manage. If we had kept a sequence by index we could have done it with difficulty.

Oh I see, that's smart! Thanks :)